arch/Kconfig

```diff
@@ -642,13 +633,14 @@ config KERNEL_VM_SIZE
 
 endif   # MMU
 
-config MEMORY_PROTECTION
```

Comment:
note: not currently seeing use-cases for CONFIG_MEMORY_PROTECTION, but if we wanted to keep it, could be something as simple as:

```
config MEMORY_PROTECTION
	bool
	default y
	depends on MMU || MPU
```